framework/src/modules/chain/chain.js

```diff
@@ -477,6 +480,20 @@ module.exports = class Chain {
 			},
 			syncInterval
 		);
+
+		/**
+		 * Periodically calculate consensus
+		 */
+		jobQueue.register(
+			'calculateConsensus',
+			async () => this._calculateConsensus(this.peers),
+			this.peers.broadhashConsensusCalculationInterval
+		);
+	}
+
+	async _calculateConsensus(peers) {
```

Comment:
It would be simple to register jobQueue in this function it self
```
jobQueue.register(
			'calculateConsensus',
			async () => {
                             const consensus = await this.peers.calculateConsensus(this.blocks.broadhash);
		              return this.logger.debug(`Broadhash consensus: ${consensus} %`);
                          },
			this.peers.broadhashConsensusCalculationInterval
		);
```

Reply:
Yeah I was thinking to do the same but from your previous comment I extracted it out, also I thought it that it might be used somewhere else too in chain.